sensor/kubernetes/certrefresh/securedcluster/certificate_requester.go

```diff
@@ -0,0 +1,132 @@
+package securedcluster
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/pkg/centralsensor"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/pkg/uuid"
+	"github.com/stackrox/rox/sensor/common/centralcaps"
+	"github.com/stackrox/rox/sensor/common/message"
+	"github.com/stackrox/rox/sensor/kubernetes/certrefresh/certificates"
+)
+
+var (
+	log = logging.LoggerForModule()
+
+	// ErrCertificateRequesterStopped is returned by RequestCertificates when the certificate
+	// requester is not initialized.
+	ErrCertificateRequesterStopped                        = errors.New("certificate requester stopped")
+	_                              certificates.Requester = (*certificateRequesterImpl)(nil)
+)
+
+// NewCertificateRequester creates a new Secured Cluster certificates requester that communicates through
+// the specified channels and initializes a new request ID for reach request.
+// To use it call Start, and then make requests with RequestCertificates, concurrent requests are supported.
+// This assumes that the returned certificate requester is the only consumer of `receiveC`.
+func NewCertificateRequester(sendC chan<- *message.ExpiringMessage,
+	receiveC <-chan *central.IssueSecuredClusterCertsResponse) certificates.Requester {
+	return &certificateRequesterImpl{
+		sendC:    sendC,
+		receiveC: receiveC,
+	}
+}
```

Comment:
I was late for the review, but this looks like a recipe for problems as it breaks some of the concurrency patterns that we try to follow. I mean in particular the rule "channels open for writing should be created and closed by the same goroutine (or object)".
Here, we accept channel for writing from the outside and need to pay special care to not write into closed channel. Ideally, we would not want to pay any special cares when writing code in the future, so maybe this could be refactored to avoid this?